source/Octopus.Server.Client/Model/RetentionPoliciesConfigurationResource.cs

```diff
@@ -0,0 +1,20 @@
+using Octopus.Client.Extensibility;
+using Octopus.Client.Extensibility.Attributes;
+
+namespace Octopus.Client.Model
+{
+    public class RetentionPoliciesConfigurationResource : IResource
+    {
+        public RetentionPoliciesConfigurationResource()
+        {
+            Id = "retentionpolicies";
+        }
+
+        public string Id { get; set; }
```

Comment:
Maybe dumb question, why is the `Id` on a configuration resource? Does this mean we can set the id?

Reply:
I'm not exactly sure but we store Configuration documents as ID, json pairs.
![image](https://github.com/OctopusDeploy/OctopusClients/assets/101079287/7ab812ea-b621-41bf-9484-2dcde3013f63)
These are all defined in server with the base class ConfigurationDocument which takes the ID as a part of the constructor. I assume the Id is just there for matching purposes and can't or at least shouldn't be changed.

Reply:
This should probably inherit from `Resource` which already contains an Id property.
If I remember correctly there should be no POST to create configuration records, so you can't just set any Id and POST it to the API, should require a GET and PUT.

Reply:
> This should probably inherit from `Resource` which already contains an Id property. If I remember correctly there should be no POST to create configuration records, so you can't just set any Id and POST it to the API, should require a GET and PUT.

The implementations are a bit all over the shop, some like ArchiveLimitConfigurationResource and EventRetentionConfigurationResource use IResource. These we're the most recent implementations of ConfigurationResources so I went with that approach. I've updated to use Resource and left the others as is, let me know you think it's worth updating those too.

Reply:
If you change them, does it change the surface area test?

Reply:
Yep, 
![image](https://github.com/OctopusDeploy/OctopusClients/assets/101079287/a94964e5-e963-43f7-8d31-99fa68854027)

Reply:
Then just leave it as is

Reply:
Only reason I ask is that configuration's ID is a db level implementation detail and is used to map between a db row and its correct configuration type, i.e. the same id always maps to the same type and vice verse. Apart from the document store nothing else should need to know about the id. So IMO exposing the id in the resource is leaky. But seems like we don't have any strict rules around this. Just raising for future consideration.